src/models/StyledComponent.js

```diff
@@ -83,7 +84,8 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
       this.attrs = Object.keys(attrs).reduce((acc, key) => {
         const attr = attrs[key]
         // eslint-disable-next-line no-param-reassign
-        acc[key] = typeof attr === 'function' ? attr(context) : attr
+        acc[key] =
+          typeof attr === 'function' && !isClass(attr) ? attr(context) : attr
```

Comment:
Instead let's use one of the methods from the `react-is` package. I think it'll work better.

Reply:
It seems that it's not possible too. With this package we can only check if `attr` is a valid element type for React but [it will accept functions too](https://github.com/facebook/react/blob/fc3777b1fe295fd2661f1974f5587d214791f04b/packages/shared/isValidElementType.js#L23). 

Also: https://github.com/facebook/react/pull/12578

Reply:
Well if we’re just concerned with classes, can we simply check if `React.Component` is in the inheritance chain for the function?

Reply:
Yeah, it's better than nothing. Will update PR.